examples/gno.land/r/gov/dao/types.gno

```diff
@@ -0,0 +1,232 @@
+package dao
+
+import (
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+type ProposalID int64
+
+func (pid ProposalID) String() string {
+	return ufmt.Sprintf("%v", int64(pid))
+}
+
+// VoteOption is the limited voting option for a DAO proposal
+// New govDAOs can create their own VoteOptions if needed in the
+// future.
+type VoteOption string
+
+const (
+	AbstainVote VoteOption = "ABSTAIN" // Side is not chosen
+	YesVote     VoteOption = "YES"     // Proposal should be accepted
+	NoVote      VoteOption = "NO"      // Proposal should be rejected
+)
+
+type VoteRequest struct {
+	Option     VoteOption
+	ProposalID ProposalID
+	Metadata   Metadata
+}
+
+type ProposalRequestBuilder struct {
+	title       string
+	description string
+	callback    func(Metadata) error
+	metadata    Metadata
+}
+
+func NewProposalRequestBuilder(title string) *ProposalRequestBuilder {
+	return &ProposalRequestBuilder{
+		title: title,
+	}
+}
+
+func (prb *ProposalRequestBuilder) WithDescription(desc string) *ProposalRequestBuilder {
+	prb.description = desc
+	return prb
+}
+
+func (prb *ProposalRequestBuilder) WithMetadata(md Metadata) *ProposalRequestBuilder {
+	prb.metadata = md
+	return prb
+}
+
+func (prb *ProposalRequestBuilder) WithCallback(cb func(Metadata) error) *ProposalRequestBuilder {
+	prb.callback = cb
+	return prb
+}
+
+func (prb *ProposalRequestBuilder) Build() ProposalRequest {
+	return ProposalRequest{
+		title:       prb.title,
+		description: prb.description,
+		callback:    prb.callback,
+		metadata:    prb.metadata,
+	}
+}
+
+type ProposalRequest struct {
+	title       string
+	description string
+	callback    func(Metadata) error
+	metadata    Metadata
+}
+
+func (p *ProposalRequest) Title() string {
+	return p.title
+}
+
+func (p *ProposalRequest) Description() string {
+	return p.description
+}
+
+func (p *ProposalRequest) Callback() func(Metadata) error {
+	return p.callback
+}
+
+func (p *ProposalRequest) Metadata() Metadata {
+	return p.metadata
+}
+
+type Proposal struct {
+	author Author
+
+	title       string
+	description string
+
```

Comment:
Isn't the idea behind metadata to provide useful information? If so, why not store it here and eventually expose a MetadataString() helper?

I still feel that we should consider removing metadata completely, but I'm trying to find ways it can be useful.

Reply:
Okay, it wasn't clear, but ExecutorString is doing exactly this.

Can we gather more opinions on whether we should keep the Metadata struct at all? I believe we shouldn't, but if we decide to keep it, perhaps we could rename it to ExecutorMetadata?